PayPalNativeCheckout/src/main/java/com/braintreepayments/api/PayPalNativeCheckoutClient.java

```diff
@@ -1,14 +1,468 @@
+                    environment = Environment.LIVE;
+                } else {
+                    callback.onResult(new IllegalArgumentException("Invalid PayPal Environment"));
+                    return;
+                }
+
+                // Start PayPalCheckout flow
+                PayPalCheckout.setConfig(
+                    new CheckoutConfig(
+                        activity.getApplication(),
+                        configuration.getPayPalClientId(),
+                        environment,
+                        payPalRequest.getNativeConfig().getReturnUrl()
+                    )
+                );
+
+                registerCallbacks(configuration, payPalRequest, payPalResponse);
+
+                PayPalCheckout.startCheckout(createOrderActions -> {
+                    if (payPalRequest instanceof PayPalCheckoutRequest) {
+                        createOrderActions.set(payPalResponse.getPairingId());
+                    } else if (payPalRequest instanceof PayPalVaultRequest) {
+                        createOrderActions.setBillingAgreementId(payPalResponse.getPairingId());
+                    } else {
+                        callback.onResult(new Exception("Unsupported action type"));
+                    }
+                });
+                callback.onResult(null);
+            } else {
+                callback.onResult(error);
+            }
+        });
+    }
+
+    private void registerCallbacks(
+        final Configuration configuration,
+        final PayPalRequest payPalRequest,
+        final PayPalResponse payPalResponse
+    ) {
+        PayPalCheckout.registerCallbacks(
+                approval -> {
+
+                    PayPalAccount payPalAccount = setupAccount(configuration, payPalRequest, payPalResponse);
+
+                    internalPayPalClient.tokenize(payPalAccount, (payPalAccountNonce, error) -> {
+                        if (payPalAccountNonce != null) {
+                            listener.onPayPalSuccess(payPalAccountNonce);
+                        } else {
+                            listener.onPayPalFailure(new Exception("PaypalAccountNonce is null"));
+                        }
+                    });
+
+                },
+                () -> listener.onPayPalFailure(new Exception("User has canceled")),
+                errorInfo -> listener.onPayPalFailure(new Exception(errorInfo.getError().getMessage()))
+        );
+    }
+
+    private PayPalAccount setupAccount(
+        final Configuration configuration,
+        final PayPalRequest payPalRequest,
+        final PayPalResponse payPalResponse
+    ) {
+        PayPalAccount payPalAccount = new PayPalAccount();
+
+        String merchantAccountId = payPalRequest.getMerchantAccountId();
+        String paymentType = payPalRequest instanceof PayPalVaultRequest ? "billing-agreement" : "single-payment";
+        payPalAccount.setClientMetadataId(configuration.getPayPalClientId());
```

Comment:
can I use this client ID? or should this be the correlationID here? @tdchow

Reply:
I think it would be what you have here. I believe the merchant's clientID would be connected to the BT account somehow, and the string value of the clientID would be returned in the BT configuration call. I may be completely off here, but please someone from BT chime in if you can!

Reply:
Yeah whichever values we're using in the PayPal web module should be fine.